src/app/content/components/Page.spec.tsx

```diff
@@ -925,14 +928,21 @@ describe('Page', () => {
 
     expect(root.querySelector('[data-testid=banner-body]')).toBeFalsy();
 
-    const highlightData = jest.spyOn(highlightUtils, 'highlightData').mockReturnValueOnce(() => undefined);
+    const highlightData = jest.spyOn(highlightUtils, 'highlightData').mockReturnValueOnce(() => ({} as any));
 
     // normally, search result selection handler would noop if the
     // search result is the same. This makes it think that a new highlight was
     // added and will force reselection
 
+    const helpers = {
+      ...createTestServices(),
+      dispatch: store.dispatch,
+      getState: store.getState,
+    };
+    const hook = (require('../highlights/hooks/createHighlight').hookBody)(helpers);
+    await hook(createHighlight({locationStrategies: [{type: 'TextPositionSelector'}]} as any, {} as any));
```

Comment:
Nor this nor the previous version of the test check what they were meant to check.

When a new highlight gets added (and it passes [this validation](https://github.com/openstax/rex-web/blob/e96ea33a943ba0280e7073612fcb353a12980238/src/app/content/components/Page/highlightManager.ts#L154) (it didnt in the previous version of the test)), [highlightsManager.update](https://github.com/openstax/rex-web/blob/e96ea33a943ba0280e7073612fcb353a12980238/src/app/content/components/Page/PageComponent.tsx#L96) returns true, which makes [onHighlightSelect](https://github.com/openstax/rex-web/blob/e96ea33a943ba0280e7073612fcb353a12980238/src/app/content/components/Page/PageComponent.tsx#L104) run again, even though no new search result was selected. This makes the search error appear again. 
To prevent it [this id was added](https://github.com/openstax/rex-web/blob/e96ea33a943ba0280e7073612fcb353a12980238/src/app/content/components/Page/PageComponent.tsx#L114) and since it's generated to be the same unless selected result, query or page changed, it would prevent the error from being shown again.

So ideally, if the id was replaced with a random string each time `onHighlightSelect` runs, the test would fail (but it doesn't since no new highlights are actually added to the store)


In retrospec similar condition to [this one](https://github.com/openstax/rex-web/blob/e96ea33a943ba0280e7073612fcb353a12980238/src/app/content/components/Page/searchHighlightManager.ts#L73) could be wrappinig the `onSelect` callback and it should solve the same problem that the id generation is solving.